### ISSUE TITLE
Feature Request: Make Triumphs->Patterns list show which vendor / currency are needed for purchasing a weapon

### ISSUE THREAD

**Issue by jchunkins** [dcaslin/d2-checklist#537]
I am constantly forgetting which weapon goes with which vendor and what currency you need to grind for to purchase the weapon. It would be great if the list in Triumphs ->Patterns could show you that information. That way I can look at the list and say "weapon X is at 80% so I just need to grind out one more weapon X using currency Y from vendor Z".

**Comment by dcaslin** [comment]
Ah do you mean like the Seraphic Umbral energy and all that jazz? (I'm struggling w/ that myself, and the related "Do I have a redbar ready to buy this week?" question)

**Comment by jchunkins** [comment]
Yep... exactly it. In Triumphs ->Patterns you're already giving part of what I need to know but not all of it. I can easily see what weapons I need to craft, and their percentage for pattern progress, but what's missing is:

- currency (e.g. `Opulent Umbral Energy`)
  - what type?
  - how much does it cost for a given weapon?
  - how much do I have available?
- vendor to purchase from
- can I purchase a pattern this week?


Just for clarity using another example:

- **Vendor**: Crown of Sorrow
  - **Cost**: 27 `Opulent Umbral Energy`
  - **In Inventory**: 10 `Opulent Umbral Energy`
  - Offers the following weapons which takes (`Opulent Umbral Energy`) to craft:
    - Haunted Weapons
      - Hollow Denial
      - Bump in the Night
      - Tears of Contrition
      - Nezarec's Whisper
      - Firefright
      - Without Remorse
    - Opulent Weapons
      - Austringer
      - Drang (Baroque)
      - Beloved
      - Calus Mini-Tool
 

So the thing I'd want displayed in the user interface somehow is the following information:
"weapon `Austringer` is at 80% pattern completion, and I have 17 `Opulent Umbral Energy`, so I just need to grind out 10 more `Opulent Umbral Energy` to reach 27 and then head to vendor `Crown of Sorrow` to purchase it".

Also to be clear, I am not asking for the UI to look like that wording. It's just the "question" in my head about what I want to know the answers to. So looking for something like this in the UI (please forgive the poor text representation here):


| Name | Progress | Desc |
|-|-|-|
| Austringer </p>Patterns & Catalysts / Primary Weapon Patterns / Hand Cannons / Austringer | 80 % </p> [progress bar]| Completing Deepsight Resonance extractions on this weapon will unlock its Pattern. Purchasable from vendor `Crown of Sorrow`. You have one weapon you can purchase from this vendor this week.</p>Objectives:</p>- 4/5 Pattern progress</p>[progress bar]</p>- 10/27 `Opulent Umbral Energy`|

Again, I have no idea if the API gives you all of this information, but if it does it would be great to see something like the above UI example.

And as long as I am being greedy about it, it would also be great if there was some link where I can see your suggestion for a god roll when it comes time to reshape a weapon. I can create a separate feature request for that idea if you'd like.
